docs/docs/sample-scripts/entra/find-obsolete-m365-groups/index.mdx

```diff
@@ -0,0 +1,293 @@
+      $Group.Reasons += $reason
+      
+      try {
+        $Global:ObsoleteGroups.Add($Group.Reference.id, $Group)
+      }
+      catch {}
+    }
+  }
+
+  function Test-SharePointActivity {
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory = $true)] [GroupInfo] $Group
+    )
+
+    function Get-ParsedDate {
+      [CmdletBinding()]
+      param (
+        [Parameter(Mandatory = $true)] [String] $JavascriptDateString
+      )
+
+      $dateParts = [regex]::Matches($JavascriptDateString, '\d+') | ForEach-Object { $_.Value }
+
+      # Convert the parts to integers
+      $year = [int]$dateParts[0]
+      $month = [int]$dateParts[1] + 1
+      $day = [int]$dateParts[2]
+      $hour = [int]$dateParts[3]
+      $minute = [int]$dateParts[4]
+      $second = [int]$dateParts[5]
+      # $millisecond = [int]$dateParts[6]
+
+      # return a DateTime object
+      $dateObject = New-Object -TypeName DateTime -ArgumentList $year, $month, $day, $hour, $minute, $second
+      $dateObject
+    }
+    
+    $WarningDate = (Get-Date).AddDays(-90)
+    
+    # Not possible to retrieve property 'LastContentModifiedDate' via command 'm365 spo site get --url $group.siteUrl', so we need to use filtering:
+    $spoSite = m365 spo site list --filter "Url -eq '$($Group.Reference.siteUrl)'" | ConvertFrom-Json
+    $spoSite.LastContentModifiedDate = Get-ParsedDate -JavascriptDateString $spoSite.LastContentModifiedDate
+    if ($spoSite.LastContentModifiedDate -lt $WarningDate) {
+      Write-Host "  → potentially obsolete (SPO last content modified: $($spoSite.LastContentModifiedDate))" -ForegroundColor Yellow
+      $reason = "Low SharePoint activity ($($spoSite.LastContentModifiedDate))"
+
+      $Group.SharePointStatus = @{
+        Reason = $reason
+      }
+      $Group.TestStatus = "🟡 Warning"
+      $Group.Reasons += $reason
+
+      try {
+        $Global:ObsoleteGroups.Add($Group.Reference.id, $Group)
+      }
+      catch {}  
+    }  
+  }
+
+  function Test-ConversationActivity {
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory = $true)] [GroupInfo] $Group
+    )
+
+    $WarningDate = (Get-Date).AddDays(-365)
+      
+    $a = m365 entra m365group conversation list --groupId $Group.Reference.id | ConvertFrom-Json | Where-Object {
+      [datetime]$_.lastDeliveredDateTime -lt $WarningDate
+    }
+    if (!$a -or $a.Length -eq 0) { return }
+
+    Write-Host "  → potentially obsolete ($($a.Length) conversation item$($a.Length -gt 1 ? 's' : '') created more than 1 year ago)" -ForegroundColor Yellow
+    $reason = "$($a.Length) conversation item$($a.Length -gt 1 ? 's' : '') created more than 1 year ago)"
+
+    $Group.MailboxStatus = @{
+      OutdatedConversations = $a | Sort-Object -Property lastDeliveredDateTime
+      Reason                = $reason
+    }
+    $Group.TestStatus = "🟡 Warning"
+    $Group.Reasons += $reason
+
+    try {
+      $Global:ObsoleteGroups.Add($Group.Reference.id, $Group)
+    }
+    catch { }
```

Comment:
Noticed you're using `try/catch` to add an item to a list quite some time. Why is this needed?

Reply:
@milanholemans, this is because I'll mutate the `$Group` object in several steps (by adding testing results). I want to make sure that an possibly obsolete group is only added once to the `ObsoleteGroups` collection. If the group is already existing, the `$Global:ObsoleteGroups.Add(...)` action will throw an error. Instead of always checking if the group was already added to the collection, I'll catch the error and disregard it (because it's intentional behavior).

In these cases, we also could write a debug message to the stdout. Would you prefer this option?

Reply:
Done!